### PR TITLE
Some fixes for sensors after testing live

### PR DESCRIPTION
**Pull request by tmjo**
Figures from Easee comes in kW and kWh, but only kW is modified to W. If this is kept, I guess kWh should be Wh as well and go through the *1000 method. I prefer to keep kW and kWh, but I don't know if this is some HA procedure to keep W and Wh?

### CODE REVIEW

File: custom_components/easee/sensor.py [hunk 1]
```diff
@@ -332,7 +354,11 @@ async def async_update(self):
             self._state = self.get_value_from_key(self._state_key)
             if self._state_func is not None:
                 charger_state = await self.charger.get_state(from_cache=True)
-                self._state = self._state_func(charger_state)
+                charger_config = await self.charger.get_config(from_cache=True)
+                try:
+                    self._state = self._state_func(charger_state)
+                except:
+                    self._state = self._state_func(charger_config)
```
Comment (fondberg) [inline custom_components/easee/sensor.py:361]: I think it is better to look at the _state_key and see if it contains "config" or "state"

File: custom_components/easee/sensor.py [hunk 2]
```diff
@@ -332,7 +354,11 @@ async def async_update(self):
             self._state = self.get_value_from_key(self._state_key)
             if self._state_func is not None:
                 charger_state = await self.charger.get_state(from_cache=True)
-                self._state = self._state_func(charger_state)
+                charger_config = await self.charger.get_config(from_cache=True)
+                try:
```
Comment (fondberg) [inline custom_components/easee/sensor.py:358]:
```suggestion
                if self._state_key.startswith("state"):
                    self._state = self._state_func(charger_state)
```

File: custom_components/easee/sensor.py (hunk 1 above)
Comment (fondberg) [inline custom_components/easee/sensor.py:361]:
```suggestion
                if self._state_key.startswith("config"):
                    self._state = self._state_func(charger_config)
```

Comment (tmjo) [reply custom_components/easee/sensor.py:361]: You're right, this was dirty from my side, getting late I guess :) My problem was that I needed it for the attribute keys as well, not only state_key. But later I changed the state to be config to, so I guess it works now.

**Comment by tmjo** [comment]
Opinions on kW/kWh vs W/Wh?

File: custom_components/easee/sensor.py [hunk 3]
```diff
@@ -61,21 +59,21 @@ def watts_to_kilowatts(value):
     "total_power": {
         "key": "state.totalPower",
         "attrs": [],
-        "units": "W",
-        "convert_units_func": watts_to_kilowatts,
+        "units": "kW",
```
Comment (fondberg) [inline custom_components/easee/sensor.py:62]: When I tested I got W for this. Are you sure you got kW?

**Comment by fondberg** [comment]
> Opinions on kW/kWh vs W/Wh?

I didn't really test other than the total which I got in W not kW

File: custom_components/easee/sensor.py (hunk 3 above)
Comment (tmjo) [reply custom_components/easee/sensor.py:62]: Quite sure, but I'll double check.

Comment (tmjo) [reply custom_components/easee/sensor.py:62]: Here's what I get when running from CLI where I've only rounded the numbers:

image

Comment (fondberg) [reply custom_components/easee/sensor.py:62]: Let's merge it and I can test again

**Pull request merged**